src/Assertion/Factory.php

```diff
@@ -1,44 +1,25 @@
 <?php
-/**
- * @see https://github.com/dotkernel/dot-rbac/ for the canonical source repository
- * @copyright Copyright (c) 2017 Apidemia (https://www.apidemia.com)
- * @license https://github.com/dotkernel/dot-rbac/blob/master/LICENSE.md MIT License
- */
 
-declare(strict_types = 1);
+declare(strict_types=1);
 
 namespace Dot\Rbac\Assertion;
 
 use Dot\Rbac\Exception\RuntimeException;
 use Psr\Container\ContainerInterface;
 
-/**
- * Class Factory
- * @package Dot\Rbac\Assertion
- */
+use function sprintf;
+
 class Factory
 {
-    /** @var  ContainerInterface */
-    protected $container;
-
-    /** @var  AssertionPluginManager */
-    protected $assertionPluginManager;
+    protected ContainerInterface $container;
+    protected ?AssertionPluginManager $assertionPluginManager;
 
-    /**
-     * Factory constructor.
-     * @param ContainerInterface $container
-     * @param AssertionPluginManager|null $assertionPluginManager
-     */
-    public function __construct(ContainerInterface $container, AssertionPluginManager $assertionPluginManager = null)
+    public function __construct(ContainerInterface $container, ?AssertionPluginManager $assertionPluginManager = null)
```

Comment:
```suggestion
    public function __construct(protected ContainerInterface $container, protected ?AssertionPluginManager $assertionPluginManager = null)
```
and then you can remove:
* the two class property definitions
* the `__constructor`'s body where the class property values are assigned